### PR TITLE
Upgrade Three.js

### PR DESCRIPTION
**Pull request by axelboc**
- https://github.com/mrdoob/three.js/releases
- https://github.com/mrdoob/three.js/wiki/Migration-Guide

### CODE REVIEW

File: apps/demo/vite.config.js [hunk 1]
```diff
@@ -14,7 +14,6 @@ export default defineConfig({
   optimizeDeps: { esbuildOptions: { target: 'es2020' } },
   build: {
     target: 'es2020',
-    // Out of memory! https://github.com/vitejs/vite/issues/2433
-    // sourcemap: true,
```
Comment (axelboc) [inline apps/demo/vite.config.js:18]: Seems to work again! :tada:

File: packages/app/jest.config.js [hunk 2]
```diff
@@ -9,10 +9,9 @@ const DEPS_TO_TRANSFORM = [
   'd3-time',
   'd3-time-format',
   'internmap',
-  'three',
```
Comment (axelboc) [inline packages/app/jest.config.js:12]: The ESM entrypoint is now properly referenced in Three's `package.json`, so this is no longer necessary.

File: apps/storybook/package.json [hunk 3]
```diff
@@ -40,7 +40,7 @@
     "@types/node": "^18.17.15",
     "@types/react": "^18.2.21",
     "@types/react-dom": "^18.2.7",
-    "@types/three": "0.141.0",
+    "@types/three": "0.156.0",
```
Comment (axelboc) [inline apps/storybook/package.json:43]: Almost forgot about the types...

File: packages/lib/src/vis/heatmap/utils.ts [hunk 4]
```diff
@@ -164,19 +164,19 @@ export function toTextureSafeNdArray(
...
 ): undefined;
 export function getDataTexture(
   values: NdArray<TextureSafeTypedArray | Uint16Array> | undefined,
-  magFilter?: TextureFilter,
+  magFilter?: MagnificationTextureFilter,
 ): DataTexture | undefined;
 export function getDataTexture(
   values: NdArray<TextureSafeTypedArray | Uint16Array> | undefined,
-  magFilter: TextureFilter = NearestFilter,
+  magFilter: MagnificationTextureFilter = NearestFilter,
```
Comment (axelboc) [inline packages/lib/src/vis/heatmap/utils.ts:179]: The type of the `magFilter` argument of the `DataTexture` is now stricter.

**Review by loichuder — APPROVED** [review]

**Pull request merged**